### DISCUSSION TITLE
data provider fresh delay when side panel insert or update

### DISCUSSION THREAD

**Discussion by WednWen** [Budibase/budibase#10296] [Help] [answered]
Hello everyone,
I have a problem when i use side panel for update data which in MYSQL table which named PlanTable. There is a page display some statistic data which from StatTable. These is a button in side panel which had defined 4 action as below:
1. Save Row is using for save the change rows
2. Execute Query is an action I define for accumulating some values which from above PlanTable, and insert into a statistic table which named StatTable in MYSQL.
3. clear form is using for clear the side panel's form
4. close side panel is using for close the side panel
The problem appeared when I testing the page, I clicked the button to update a row. PlanTable and StatTable both update, but page has not change. then i update another row once more, both tables are updated, then Something confusing has emerged. data in the page changed, but display the previous one. Eg, initial value is 10, after first update StatTable recorded 15, page still display 10. After second update, StatTable recorded 20, but page display 15.
I think it cause by execute sequence about button clicked and automatic refresh data provider, they look like Synchronous. But if I want the data in page fresh after the last action of the button( close side panel), how can I do it.

**Comment by WednWen** [comment]
![page](https://user-images.githubusercontent.com/130734446/231972425-7648efda-ade2-43d6-8560-5824117cbb10.png)
page have a data provider from StatTable, and text field got data from the data provider.
![side panel](https://user-images.githubusercontent.com/130734446/231972828-bde26a5e-2134-4231-8504-822eb16f6cb8.png)
side panel define 4 action as above

**Reply by ConorWebb96** [reply, marked as answer]
Hey @WednWen,

Have you tried using the Refresh data provider action?

Screenshots below:
<img width="1290" alt="Refresh data provider" src="https://user-images.githubusercontent.com/126772285/232019988-29d97268-ea18-4966-ac8d-c1c1b15a868d.png">
<img width="1285" alt="Added action" src="https://user-images.githubusercontent.com/126772285/232019990-b071cc4d-fbda-45d1-82fa-78556a5fd25a.png">

I hope this is what you are looking for!

**Reply by WednWen** [reply]
Hey @ConorWebb96,
thanks for your answer.
I used to try the method what you mentioned, but just one data provider i could by chose is a data provider(GQProjectDetailDP) in the side panel,  is not the one(PlanBaseDP) i want to asynchronous refresh. is there any progress for my operation?
![DataPrivoder](https://user-images.githubusercontent.com/130734446/232181165-a8741c9b-838b-41b5-a686-514acbd39946.png)

**Reply by ConorWebb96** [reply]
Is there a reason why you can have the side panels inside the PlanBaseDP? 
Doing this would give you access to the Data Provider you wish to refresh.

I hope this makes sense, if it doesn't work let us know and we can look for another workaround!

**Reply by WednWen** [reply]
Dear conorWebb96,
Thanks for your answer, i make a mistake when i edit the page. The side panel could be make into the first data provider.
After that it refreshed, thanks a lot.
